management/src/main/java/io/micronaut/management/health/indicator/threads/OnlyWhenRunningOnTheJvmCondition.java

```diff
@@ -0,0 +1,50 @@
+/*
+ * Copyright 2017-2024 original authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.micronaut.management.health.indicator.threads;
+
+import io.micronaut.context.condition.Condition;
+import io.micronaut.context.condition.ConditionContext;
+import io.micronaut.core.annotation.Internal;
+
+/**
+ * Condition to hide parts of an application that only work when running on the JVM.
+ * Internal implementation is identical to {@code if (!ImageInfo.inImageCode()).
+ * @author Sergio del Amo
+ * @since 4.8.0
+ */
+@Internal
+class OnlyWhenRunningOnTheJvmCondition implements Condition {
```

Comment:
seems incorrectly named, the name should be `OnlyWhenRunningOnNativeImage`

Reply:
this seems like it belongs in inject or context somewhere not in management

Reply:
The name is correct. The condition is meant to load a bean only on the jvm not in native.

Reply:
oh I see

Reply:
> this seems like it belongs in inject or context somewhere not in management

I moved to inject and [renamed to RunningOnTheJvm](https://github.com/micronaut-projects/micronaut-core/pull/11388/commits/6d6b3bff612b96463953c8af522333535cc637b5)